### PR TITLE
fix: address issue with computeclass tables not converting properly

### PR DESCRIPTION
**Pull request by tylerslaton**
This fixes an issue where `acorn offerings computeclasses` and `kubectl get clustercomputeclasses` would fail with the following error:

```
  ✗  ERROR:  template: :1:39: executing "" at <memoryToRange .Memory>: error calling memoryToRange: object passed is not a ComputeClassMemory struct
```

This was due to a recent change where the `Memory` struct on `apiv1.ComputeClass` and `adminv1.ClusterComputeClass` are now different. The fix is to check for both types and process them accordingly when converting values into a table.

### Checklist
- [X] The title of this PR would make a good line in Acorn's Release Note's Changelog
- [X] The title of this PR ends with a link to the main issue being address in parentheses, like: `This is a title (#1216)`. [Here's an example](https://github.com/acorn-io/runtime/pull/1199)
- [X] All relevant issues are referenced in the PR description. *NOTE: don't use [GitHub keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) that auto-close issues*
- [X] Commits follow [contributing guidance](https://github.com/acorn-io/runtime/blob/main/CONTRIBUTING.md#commits)
- [X] Automated tests added to cover the changes. If tests couldn't be added, an explanation is provided in the Verification and Testing section
- [X] Changes to user-facing functionality, API, CLI, and upgrade impacts are clearly called out in PR description
- [ ] PR has at least two approvals before merging (or a reasonable exception, like it's just a docs change)

### CODE REVIEW

File: pkg/cli/builder/table/funcs.go [hunk 1]
```diff
@@ -272,46 +272,58 @@ func DisplayRange(minVal, maxVal any) (string, error) {
...
+		return defaultMemory(b.Default, b.Max), nil
+	default:
+		return "", fmt.Errorf("object passed cannot be converted into a ComputeClassMemory struct")
 	}
+}
 
-	result := b.Default
-	if b.Default == "0" || b.Default == "" {
-		result = b.Max
+func defaultMemory(def, max string) string {
```
Comment (g-linville) [inline pkg/cli/builder/table/funcs.go:285]: This function looks incorrect. `def` is always returned instead of `result`.

**Review by g-linville — CHANGES_REQUESTED** [review]

File: pkg/cli/builder/table/funcs.go (hunk 1 above)
Comment (tylerslaton) [reply pkg/cli/builder/table/funcs.go:285]: Yep, you're right. Fixed.

File: pkg/controller/appdefinition/testdata/computeclass/all-set/existing.yaml [hunk 2]
```diff
@@ -5,10 +5,6 @@ metadata:
   name: sample-compute-class
 description: Simple description for a simple ComputeClass
 cpuScaler: 0.25
-memory:
-  min: 1Mi
-  max: 2Mi
-  default: 1Mi
```
Comment (thedadams) [inline pkg/controller/appdefinition/testdata/computeclass/all-set/existing.yaml:11]: Why is this changing? Seems like we were trying to test something here.

Comment (tylerslaton) [reply pkg/controller/appdefinition/testdata/computeclass/all-set/existing.yaml:11]: It shouldn't be, just a typo left over it would seem.

Comment (tylerslaton) [reply pkg/controller/appdefinition/testdata/computeclass/all-set/existing.yaml:11]: Thought I caught all of this already, my bad.

**Review by thedadams — APPROVED** [review]

File: pkg/cli/builder/table/funcs.go [hunk 3]
```diff
@@ -272,46 +272,58 @@ func DisplayRange(minVal, maxVal any) (string, error) {
...
+	var result string
```
Comment (g-linville) [inline pkg/cli/builder/table/funcs.go:286]: This needs to be
```suggestion
	result := def
```

**Review by g-linville — CHANGES_REQUESTED** [review]

File: pkg/cli/builder/table/funcs.go (hunk 3 above)
Comment (tylerslaton) [reply pkg/cli/builder/table/funcs.go]: Thank you. This has been fixed.

**Comment by cjellick** [comment]
go ahead and merge. you addressed @Grant Linville's comment and he's getting lunch. donnie approved. im good with you merging it

**Review by cjellick — APPROVED** [review]

**Comment by tylerslaton** [comment]
https://github.com/acorn-io/runtime/issues/2429

**Pull request merged**